packages/dcos-integration-test/extra/test_calico_networking.py

```diff
@@ -0,0 +1,190 @@
+import contextlib
+import logging
+import uuid
+from abc import ABC, abstractmethod
+from ipaddress import IPv4Address, IPv4Network
+
+import pytest
+import test_helpers
+
+from dcos_test_utils import marathon
+from test_networking import ensure_routable, MarathonApp, MarathonPod
+
+
+__maintainer__ = 'mainred'
+__contact__ = 'networking-team@mesosphere.io'
+
+log = logging.getLogger(__name__)
+
+
+class CalicoNetworkTestBase(ABC):
+    CALICO_NETWORK_NAME = 'calico'
+    NETWORK_TYPE = marathon.Network.USER
+
+    @abstractmethod
+    def get_container_type(self):
+        return None
+
+    @pytest.fixture(scope="class")
+    def slaves(self, dcos_api_session):
+        return dcos_api_session.slaves + dcos_api_session.public_slaves
+
+    @pytest.fixture(scope="class")
+    def expanded_config(self):
+        expanded_config = test_helpers.get_expanded_config()
+        return expanded_config
+
+    def get_calico_app(self, host):
+        """Returns a calico network application definition with unique id """
+        network_name = self.CALICO_NETWORK_NAME
+        container_type = self.get_container_type()
+        network_type = self.NETWORK_TYPE
+        # NOTE: tasks created by marathon in Enterprise DC/OS will be attached
+        # with a label DCOS_SPACE: task_id. And calico CNI restricts the length
+        # of label value in 63 characters.
+        test_uuid = uuid.uuid4().hex[:5]
+        app_name_fmt = "/integration-test/calico/app-{}-{}-{}".format(
+            str(container_type).lower(), host, test_uuid)
+        marathon_app = MarathonApp(
+            container=container_type,
+            network=network_type,
+            host=host,
+            network_name=network_name,
+            app_name_fmt=app_name_fmt)
+        return marathon_app
+
+    @pytest.fixture(scope="class")
+    def mesos_calico_apps(self, dcos_api_session, slaves):
+        """Returns mesos calico app object in healthy state
+
+        Three applications will be returned if there are more than one slave in
```

Comment:
Why do we have such a variance? Do the integration tests run this test in two different environments?

Reply:
No, because test_networking is filled with bug fixes, this helps me maintain the code later.